api/geth_backend.go

```diff
@@ -886,8 +901,24 @@ func (b *GethStatusBackend) injectAccountIntoServices() error {
 		return err
 	}
 
-	if whisperService != nil {
-		st, err := b.statusNode.ShhExtService()
+	wakuService, err := b.statusNode.WakuService()
+
+	switch err {
+	case node.ErrServiceUnknown: // Whisper was never registered
+	case nil:
+		if err := wakuService.DeleteKeyPairs(); err != nil { // err is not possible; method return value is incorrect
+			return err
+		}
+		b.selectedAccountShhKeyID, err = wakuService.AddKeyPair(identity)
```

Comment:
Let's changed it to `selectedAccountKeyID`. We should also make sure it's not possible for a client to have both Whisper and Waku started. It's fine for the relay node.